src/java/org/apache/cassandra/locator/DynamicEndpointSnitch.java

```diff
@@ -210,8 +368,10 @@ public String getDatacenter(InetAddressAndPort endpoint)
 
         // TODO: avoid copy
         replicas = subsnitch.sortedByProximity(address, replicas);
-        HashMap<InetAddressAndPort, Double> scores = this.scores; // Make sure the score don't change in the middle of the loop below
-                                                           // (which wouldn't really matter here but its cleaner that way).
+        // Make sure the score don't change in the middle of the loop below
+        // (which wouldn't really matter here but its cleaner that way).
+        final Map<InetAddressAndPort, Double> scores = this.scores;
```

Comment:
I don't think we generally do final on the stack or for parameters to functions. I can't find anything in the style guide that says you can't do it, but I recall being told that at one point. I'll ask around.

Reply:
Yea I was somewhat surprised to see that too, I assumed there was a reason that it was final in `sortedByProximityWithBadness` and just made `sortedByProximityWithScores` consistent, but tbh I don't care and will switch to whichever way is better for you ;-)

Reply:
I asked and yes we do strongly discourage final on the stack and on method parameters.

Reply:
Cool, I made them consistent the other way.